python/parquet_tools/common/schema.py

```diff
@@ -0,0 +1,700 @@
+# This file is part of parquet_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This module encapsulates all the code to process schema files and provide
+support for schemas that allow null values for non-floating point values.
+
+This module augments schemas that allow null values for data types which
+pandas does not support to be null. Specifically, for each such column it
+adds a  column with a name of '<cname>_ISNULL', where <cname> is the name
```

Comment:
it would probably be cooler if the modifier was configurable but if this serves the need then whatever don't do it 🙃

Reply:
Yeah, it serves he need. We'll keep it this way until there are enough complaints which I doubt there will be.